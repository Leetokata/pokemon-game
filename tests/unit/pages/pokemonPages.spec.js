import { shallowMount, mount } from '@vue/test-utils' 
import  pokemonPages  from "@/pages/PokemonPages";
import { pokemons } from '../mocks/pokemons.mock'

describe('Pokemon Page Component', ()=>{

    let wrapper

    beforeEach(()=>{
        
        wrapper = shallowMount( pokemonPages )
    })
    
    test('debe de llamar mixPokemonArray al montar', ()=>{

        const mixPokemonArraySpy = jest.spyOn( pokemonPages.methods, 'mixPokemonArray')
        wrapper = shallowMount( pokemonPages )
        
        expect( mixPokemonArraySpy ).toHaveBeenCalled()
        
    })
    
    test('debe hacer match con el snapshot cuando cargan los pokemons', ()=>{
        const wrapper = shallowMount( pokemonPages, {
            data() {
                return {
                    pokemonArr : pokemons,
                    pokemon:  pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: '',
                }
            }
        } )

        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('debe de mostrar los componentes PokemonsPicture y PokemonOptions', ()=>{
        const wrapper = shallowMount( pokemonPages, {
            data() {
                return {
                    pokemonArr : pokemons,
                    pokemon:  pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: '',
                }
            }
        } )

        const pokemonsPicture = wrapper.find('pokemon-picture-stub')
        const pokemonsOptions = wrapper.find('pokemon-options-stub')

        expect( pokemonsPicture.exists() ).toBe(true)
        expect( pokemonsOptions.exists() ).toBe(true)
        expect( pokemonsPicture.attributes('pokemonid')).toBe('501')
        expect( pokemonsOptions.attributes('pokemons')).toBeTruthy()
    })

    test('pruebas con checkAnswer', async()=>{
        const wrapper = shallowMount( pokemonPages, {
            data() {
                return {
                    pokemonArr : pokemons,
                    pokemon:  pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: '',
                }
            }
        } )

        await wrapper.vm.checkAnswer(501)

        expect( wrapper.find('h2').exists()).toBeTruthy()
        expect( wrapper.vm.showPokemon ).toBe(true)
        expect( wrapper.find('h2').text()).toBe(`Correcto!, ${pokemons[0].name}`)

        await wrapper.vm.checkAnswer(58)

       // console.log(wrapper.find('h2').text())

        expect( wrapper.vm.message ).toBe(`Oops!, era ${pokemons[0].name}`)
    })


})