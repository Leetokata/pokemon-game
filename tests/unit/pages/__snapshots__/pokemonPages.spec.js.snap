// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Page Component debe hacer match con el snapshot cuando cargan los pokemons 1`] = `
<div>
  <h1>Â¿Quien es este Pokemon?</h1><!-- Imagen -->
  <pokemon-picture-stub pokemonid="501" showpokemon="false"></pokemon-picture-stub><!-- Opciones -->
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
